packages/mui-material/src/ButtonBase/ButtonBase.js

```diff
@@ -68,6 +68,21 @@ export const ButtonBaseRoot = styled('button', {
   },
 });
 
+function useRippleHandler(ripple, rippleAction, eventCallback, skipRippleAction = false) {
+  return useEventCallback((event) => {
+    if (eventCallback) {
+      eventCallback(event);
+    }
+
+    const ignore = skipRippleAction;
+    if (!ignore) {
+      ripple[rippleAction](event);
+    }
+
+    return true;
+  });
+}
+
```

Comment:
Move this function at the end, by convention helper functions should be last, and the main components/hooks at the start.